cylc/flow/pathutil.py

```diff
@@ -196,7 +215,7 @@ def make_symlink(path: Union[Path, str], target: Union[Path, str]) -> bool:
     path = Path(path)
     target = Path(target)
     if path.exists():
-        if path.is_symlink() and path.samefile(target):
+        if path.is_symlink() and target.exists() and path.samefile(target):
```

Comment:
No, it is not redundant... consider the case where a user sets up a symlink, different to that of what is in the `global.cylc [install] [symlink dirs]` section and then runs `cylc play`. In this scenario...
- `path.exists()` returns True
- `path.is_symlink()` returns True
- `path.samefile(target)` results in error with traceback, since that is the default for `samefile` if either path or target do not exist.
In this case, we want the skipping to be logged and it to pass through without error.

Reply:
Ah ok, might be worth a comment in the code

Reply:
Thanks, comment added.